File: src/main/java/com/efedotov/vaultly/dto/user/UpdateUserRequest.java
package com.efedotov.vaultly.dto.user;

import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class UpdateUserRequest {
    @Size(min = 3, max = 50)
    @Pattern(regexp = "^[a-zA-Z0-9_.-]+$", message = "Username: a-z, A-Z, 0-9, _, ., -")
    private String username;

    @Size(max = 100)
    private String firstName;

    @Size(max = 100)
    private String lastName;
}

File: lib/server/model/user/update_user_request.dart
class UpdateUserRequest {
  final String? username;
  final String? firstName;
  final String? lastName;

  UpdateUserRequest({this.username, this.firstName, this.lastName});

  factory UpdateUserRequest.fromJson(Map<String, dynamic> json) {
    return UpdateUserRequest(
      username: json['username'] as String?,
      firstName: json['firstName'] as String?,
      lastName: json['lastName'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    if (username != null) 'username': username,
    if (firstName != null) 'firstName': firstName,
    if (lastName != null) 'lastName': lastName,
  };
}
